pkg/minikube/bootstrapper/images/images.go

```diff
@@ -18,25 +18,37 @@ limitations under the License.
 package images
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"net/http"
 	"path"
 
+	"k8s.io/klog/v2"
+
 	"k8s.io/minikube/pkg/minikube/constants"
 
 	"github.com/blang/semver/v4"
 
 	"k8s.io/minikube/pkg/version"
 )
 
+const (
+	// builds a docker v2 repository API call in the format https://k8s.gcr.io/v2/coredns/coredns/tags/list
+	tagURLTemplate = "https://%s/v2/%s/tags/list"
+)
+
 // Pause returns the image name to pull for a given Kubernetes version
 func Pause(v semver.Version, mirror string) string {
 	// Note: changing this logic requires bumping the preload version
 	// Should match `PauseVersion` in:
 	// https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/constants/constants.go
 	// https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/constants/constants_unix.go
 	pv := "3.6"
-	majorMinorVersion := fmt.Sprintf("v%d.%d", v.Major, v.Minor)
 	imageName := "pause"
+	pv = findLatestTagFromRepository(fmt.Sprintf(tagURLTemplate, kubernetesRepo(mirror), imageName), pv)
```

Comment:
I think it would make sense to move this to the `else` case if we don't find the image in `constants.KubeadmImages` because right now we're committing to a network call which may be overwritten anyway. This case makes more sense to be the fallback if not found in the constants

Reply:
Agreed. Fixed.